File: versao-java/src/test/java/api/produtos/java/assertions/ProdutoAssert.java
package api.produtos.java.assertions;

import api.produtos.java.entity.Produto;

import java.math.BigDecimal;

import static org.assertj.core.api.Assertions.assertThat;

public class ProdutoAssert {
    private Produto actual;

    public ProdutoAssert(Produto actual) {
        this.actual = actual;
    }

    public ProdutoAssert hasId(Long expectedId) {
        assertThat(actual.getId()).isEqualTo(expectedId);
        return this;
    }

    public ProdutoAssert hasNome(String expectedNome) {
        assertThat(actual.getNome()).isEqualTo(expectedNome);
        return this;
    }

    public ProdutoAssert hasPreco(BigDecimal expectedPreco) {
        assertThat(actual.getPreco()).isEqualTo(expectedPreco);
        return this;
    }
}

File: versao-kotlin/src/test/kotlin/api/produtos/kotlin/assertions/ProdutoAssert.kt
package api.produtos.kotlin.assertions

import api.produtos.kotlin.entity.Produto
import org.assertj.core.api.Assertions.assertThat
import java.math.BigDecimal

class ProdutoAssert(val actual: Produto?) {
    fun hasId(expectedId: Long) =
        apply { assertThat(actual?.id).isEqualTo(expectedId) }

    fun hasNome(expectedNome: String) =
        apply { assertThat(actual?.nome).isEqualTo(expectedNome) }

    fun hasPreco(expectedPreco: BigDecimal) =
        apply { assertThat(actual?.preco).isEqualTo(expectedPreco) }
}